### ISSUE TITLE
Fixing UndefVarError: JULIA_HOME not defined

### ISSUE THREAD

**Issue by fvi21** [JuliaLang/IJulia.jl#658]
As I read in the other post there's a fix for the "JULIA_HOME not defined" problem but I don't understand very well how to repair it.
I'll share the report:
```
┌ Error: ------------------------------------------------------------
│ # Build failed for IJulia
│   exception =
│    LoadError: UndefVarError: JULIA_HOME not defined
│    Stacktrace:
│     [1] uv_error at ./libuv.jl:80 [inlined]
│     [2] unlink(::String) at ./file.jl:549
│     [3] #rm#9(::Bool, ::Bool, ::Function, ::String) at ./file.jl:151
│     [4] #rm at ./<missing>:0 [inlined]
│     [5] #installkernel#2(::String, ::Function, ::String) at /home/user/.julia/                                                       v0.7/IJulia/deps/kspec.jl:117
│     [6] installkernel(::String) at /home/user/.julia/v0.7/IJulia/deps/kspec.jl                                                       :36
│     [7] top-level scope at /home/user/.julia/v0.7/IJulia/deps/build.jl:88
│     [8] include at ./boot.jl:314 [inlined]
│     [9] include_relative(::Module, ::String) at ./loading.jl:1067
│     [10] include at ./sysimg.jl:29 [inlined]
│     [11] include(::String) at ./loading.jl:1101
│     [12] top-level scope
│     [13] eval at ./boot.jl:317 [inlined]
│     [14] eval at ./sysimg.jl:74 [inlined]
│     [15] evalfile(::String, ::Array{String,1}) at ./loading.jl:1096 (repeats 2                                                        times)
│     [16] #4 at ./none:13 [inlined]
│     [17] cd(::getfield(Main, Symbol("##4#6")){String}, ::String) at ./file.jl:                                                       72
│     [18] (::getfield(Main, Symbol("##3#5")))(::IOStream) at ./none:12
│     [19] #open#320(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(                                                       ),Tuple{}}}, ::Function, ::getfield(Main, Symbol("##3#5")), ::String, ::Vararg{S                                                       tring,N} where N) at ./iostream.jl:369
│     [20] open(::Function, ::String, ::String) at ./iostream.jl:367
│     [21] top-level scope
│     [22] eval at ./boot.jl:317 [inlined]
│     [23] eval(::Module, ::Expr) at ./sysimg.jl:74
│     [24] exec_options(::Base.JLOptions) at ./client.jl:304
│     [25] _start() at ./client.jl:455
│    in expression starting at /home/user/.julia/v0.7/IJulia/deps/build.jl:117
└ @ Main none:16
```
I already have a Jupyter Notebook running, and it's latest release.
The version of the Julia installed in my Ubuntu Xenial is:
`julia version 0.7.0-DEV`

The question here is, Can someone, please explain how to fix the error? I'm new in this but I want to learn about. My main question is what commands or changes I have to do and the order of execute them to successfully fix the error.
Thanks in advance! Regards,

Franco.

EDIT 1:
```
julia> versioninfo()
Julia Version 0.7.0-DEV.5007
Commit ee2211c (2018-05-04 21:10 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Pentium(R) CPU G4560 @ 3.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, skylake)
Environment:

```

**Comment by paulathomas** [comment]
I had this too on xubunntu 18.04 (bionic beaver). I fixed it by typing "export JULIA_HOME="<path to the directory where julia binary is>". Following which I got this:

INFO: Found Jupyter version 4.4.0: jupyter

signal (11): Segmentation fault
in expression starting at /home/paula/.julia/v0.7/IJulia/deps/kspec.jl:6
is_ast_node at /home/paula/julia/src/dump.c:445 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:499
jl_serialize_value_ at /home/paula/julia/src/dump.c:467 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:640
jl_serialize_value_ at /home/paula/julia/src/dump.c:467 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:640
jl_serialize_value_ at /home/paula/julia/src/dump.c:328 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:622
jl_serialize_value_ at /home/paula/julia/src/dump.c:2302 [inlined]
jl_compress_ast at /home/paula/julia/src/dump.c:2304
jl_method_set_source at /home/paula/julia/src/method.c:528 [inlined]
jl_new_method at /home/paula/julia/src/method.c:587 [inlined]
jl_method_def at /home/paula/julia/src/method.c:761
eval_methoddef at /home/paula/julia/src/interpreter.c:310
eval_body at /home/paula/julia/src/interpreter.c:666
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b81395a4f)
unknown function (ip: 0x11)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_parse_eval_all at /home/paula/julia/src/ast.c:852
jl_load at /home/paula/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29 [inlined]
include at ./loading.jl:1101
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_stmt_value at /home/paula/julia/src/interpreter.c:367 [inlined]
eval_body at /home/paula/julia/src/interpreter.c:700
eval_body at /home/paula/julia/src/interpreter.c:624
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b7faa9daf)
unknown function (ip: 0xb5)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_parse_eval_all at /home/paula/julia/src/ast.c:852
jl_load at /home/paula/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29 [inlined]
include at ./loading.jl:1101
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_body at /home/paula/julia/src/interpreter.c:559
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b81575f8f)
unknown function (ip: 0xffffffffffffffff)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:802
jl_toplevel_eval_in at /home/paula/julia/src/builtins.c:624
eval at ./boot.jl:317 [inlined]
eval at ./sysimg.jl:74 [inlined]
evalfile at ./loading.jl:1096
evalfile at ./loading.jl:1096 [inlined]
#4 at ./none:13 [inlined]
cd at ./file.jl:72
#3 at ./none:12
#open#320 at ./iostream.jl:369
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
open at ./iostream.jl:367
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_body at /home/paula/julia/src/interpreter.c:559
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b80308f0f)
unknown function (ip: 0xb)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:802
jl_toplevel_eval_in at /home/paula/julia/src/builtins.c:624
eval at ./boot.jl:317 [inlined]
eval at ./sysimg.jl:74
jl_apply_generic at /home/paula/julia/src/gf.c:2167
exec_options at ./client.jl:304
_start at ./client.jl:455
jl_apply_generic at /home/paula/julia/src/gf.c:2167
jl_apply at /home/paula/julia/ui/../src/julia.h:1540 [inlined]
true_main at /home/paula/julia/ui/repl.c:109
main at /home/paula/julia/ui/repl.c:240
__libc_start_main at /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310
_start at /home/paula/julia/usr/bin/julia (unknown line)
Allocations: 6855885 (Pool: 6853130; Big: 2755); GC: 16
ERROR: Build process failed.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] build!(::Array{String,1}, ::Set{Any}, ::String) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:630
 [3] (::getfield(Pkg.Entry, Symbol("##51#52")){Array{String,1},Dict{Any,Any},Set{Any}})(::String, ::IOStream) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:636
 [4] mktemp(::getfield(Pkg.Entry, Symbol("##51#52")){Array{String,1},Dict{Any,Any},Set{Any}}, ::String) at ./file.jl:425
 [5] mktemp at ./file.jl:423 [inlined]
 [6] build! at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:635 [inlined] (repeats 2 times)
 [7] build(::Array{String,1}) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:647
 [8] #4 at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:36 [inlined]
 [9] cd(::getfield(Pkg.Dir, Symbol("##4#7")){Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},typeof(Pkg.Entry.build),Tuple{Array{String,1}}}, ::String) at ./file.jl:72
 [10] #cd#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:36
 [11] cd(::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:25
 [12] build(::String, ::Vararg{String,N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/Pkg.jl:256
 [13] top-level INFO: Found Jupyter version 4.4.0: jupyter

signal (11): Segmentation fault
in expression starting at /home/paula/.julia/v0.7/IJulia/deps/kspec.jl:6
is_ast_node at /home/paula/julia/src/dump.c:445 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:499
jl_serialize_value_ at /home/paula/julia/src/dump.c:467 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:640
jl_serialize_value_ at /home/paula/julia/src/dump.c:467 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:640
jl_serialize_value_ at /home/paula/julia/src/dump.c:328 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:622
jl_serialize_value_ at /home/paula/julia/src/dump.c:2302 [inlined]
jl_compress_ast at /home/paula/julia/src/dump.c:2304
jl_method_set_source at /home/paula/julia/src/method.c:528 [inlined]
jl_new_method at /home/paula/julia/src/method.c:587 [inlined]
jl_method_def at /home/paula/julia/src/method.c:761
eval_methoddef at /home/paula/julia/src/interpreter.c:310
eval_body at /home/paula/julia/src/interpreter.c:666
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b81395a4f)
unknown function (ip: 0x11)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_parse_eval_all at /home/paula/julia/src/ast.c:852
jl_load at /home/paula/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29 [inlined]
include at ./loading.jl:1101
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_stmt_value at /home/paula/julia/src/interpreter.c:367 [inlined]
eval_body at /home/paula/julia/src/interpreter.c:700
eval_body at /home/paula/julia/src/interpreter.c:624
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b7faa9daf)
unknown function (ip: 0xb5)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_parse_eval_all at /home/paula/julia/src/ast.c:852
jl_load at /home/paula/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29 [inlined]
include at ./loading.jl:1101
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_body at /home/paula/julia/src/interpreter.c:559
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b81575f8f)
unknown function (ip: 0xffffffffffffffff)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:802
jl_toplevel_eval_in at /home/paula/julia/src/builtins.c:624
eval at ./boot.jl:317 [inlined]
eval at ./sysimg.jl:74 [inlined]
evalfile at ./loading.jl:1096
evalfile at ./loading.jl:1096 [inlined]
#4 at ./none:13 [inlined]
cd at ./file.jl:72
#3 at ./none:12
#open#320 at ./iostream.jl:369
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
open at ./iostream.jl:367
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_body at /home/paula/julia/src/interpreter.c:559
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b80308f0f)
unknown function (ip: 0xb)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:802
jl_toplevel_eval_in at /home/paula/julia/src/builtins.c:624
eval at ./boot.jl:317 [inlined]
eval at ./sysimg.jl:74
jl_apply_generic at /home/paula/julia/src/gf.c:2167
exec_options at ./client.jl:304
_start at ./client.jl:455
jl_apply_generic at /home/paula/julia/src/gf.c:2167
jl_apply at /home/paula/julia/ui/../src/julia.h:1540 [inlined]
true_main at /home/paula/julia/ui/repl.c:109
main at /home/paula/julia/ui/repl.c:240
__libc_start_main at /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310
_start at /home/paula/julia/usr/bin/julia (unknown line)
Allocations: 6855885 (Pool: 6853130; Big: 2755); GC: 16
ERROR: Build process failed.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] build!(::Array{String,1}, ::Set{Any}, ::String) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:630
 [3] (::getfield(Pkg.Entry, Symbol("##51#52")){Array{String,1},Dict{Any,Any},Set{Any}})(::String, ::IOStream) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:636
 [4] mktemp(::getfield(Pkg.Entry, Symbol("##51#52")){Array{String,1},Dict{Any,Any},Set{Any}}, ::String) at ./file.jl:425
 [5] mktemp at ./file.jl:423 [inlined]
 [6] build! at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:635 [inlined] (repeats 2 times)
 [7] build(::Array{String,1}) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:647
 [8] #4 at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:36 [inlined]
 [9] cd(::getfield(Pkg.Dir, Symbol("##4#7")){Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},typeof(Pkg.Entry.build),Tuple{Array{String,1}}}, ::String) at ./file.jl:72
 [10] #cd#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:36
 [11] cd(::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:25
 [12] build(::String, ::Vararg{String,N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/Pkg.jl:256
 [13] top-level scope

INFO: Found Jupyter version 4.4.0: jupyter

signal (11): Segmentation fault
in expression starting at /home/paula/.julia/v0.7/IJulia/deps/kspec.jl:6
is_ast_node at /home/paula/julia/src/dump.c:445 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:499
jl_serialize_value_ at /home/paula/julia/src/dump.c:467 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:640
jl_serialize_value_ at /home/paula/julia/src/dump.c:467 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:640
jl_serialize_value_ at /home/paula/julia/src/dump.c:328 [inlined]
jl_serialize_value_ at /home/paula/julia/src/dump.c:622
jl_serialize_value_ at /home/paula/julia/src/dump.c:2302 [inlined]
jl_compress_ast at /home/paula/julia/src/dump.c:2304
jl_method_set_source at /home/paula/julia/src/method.c:528 [inlined]
jl_new_method at /home/paula/julia/src/method.c:587 [inlined]
jl_method_def at /home/paula/julia/src/method.c:761
eval_methoddef at /home/paula/julia/src/interpreter.c:310
eval_body at /home/paula/julia/src/interpreter.c:666
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b81395a4f)
unknown function (ip: 0x11)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_parse_eval_all at /home/paula/julia/src/ast.c:852
jl_load at /home/paula/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29 [inlined]
include at ./loading.jl:1101
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_stmt_value at /home/paula/julia/src/interpreter.c:367 [inlined]
eval_body at /home/paula/julia/src/interpreter.c:700
eval_body at /home/paula/julia/src/interpreter.c:624
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b7faa9daf)
unknown function (ip: 0xb5)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_parse_eval_all at /home/paula/julia/src/ast.c:852
jl_load at /home/paula/julia/src/toplevel.c:894
include at ./boot.jl:314 [inlined]
include_relative at ./loading.jl:1067
include at ./sysimg.jl:29 [inlined]
include at ./loading.jl:1101
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_body at /home/paula/julia/src/interpreter.c:559
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b81575f8f)
unknown function (ip: 0xffffffffffffffff)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:802
jl_toplevel_eval_in at /home/paula/julia/src/builtins.c:624
eval at ./boot.jl:317 [inlined]
eval at ./sysimg.jl:74 [inlined]
evalfile at ./loading.jl:1096
evalfile at ./loading.jl:1096 [inlined]
#4 at ./none:13 [inlined]
cd at ./file.jl:72
#3 at ./none:12
#open#320 at ./iostream.jl:369
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
open at ./iostream.jl:367
jl_fptr_trampoline at /home/paula/julia/src/gf.c:1838
jl_apply_generic at /home/paula/julia/src/gf.c:2167
do_call at /home/paula/julia/src/interpreter.c:324
eval_value at /home/paula/julia/src/interpreter.c:416
eval_body at /home/paula/julia/src/interpreter.c:559
jl_interpret_toplevel_thunk_callback at /home/paula/julia/src/interpreter.c:798
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f7b80308f0f)
unknown function (ip: 0xb)
jl_interpret_toplevel_thunk at /home/paula/julia/src/interpreter.c:807
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:856
jl_toplevel_eval_flex at /home/paula/julia/src/toplevel.c:802
jl_toplevel_eval_in at /home/paula/julia/src/builtins.c:624
eval at ./boot.jl:317 [inlined]
eval at ./sysimg.jl:74
jl_apply_generic at /home/paula/julia/src/gf.c:2167
exec_options at ./client.jl:304
_start at ./client.jl:455
jl_apply_generic at /home/paula/julia/src/gf.c:2167
jl_apply at /home/paula/julia/ui/../src/julia.h:1540 [inlined]
true_main at /home/paula/julia/ui/repl.c:109
main at /home/paula/julia/ui/repl.c:240
__libc_start_main at /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310
_start at /home/paula/julia/usr/bin/julia (unknown line)
Allocations: 6855885 (Pool: 6853130; Big: 2755); GC: 16
ERROR: Build process failed.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] build!(::Array{String,1}, ::Set{Any}, ::String) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:630
 [3] (::getfield(Pkg.Entry, Symbol("##51#52")){Array{String,1},Dict{Any,Any},Set{Any}})(::String, ::IOStream) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:636
 [4] mktemp(::getfield(Pkg.Entry, Symbol("##51#52")){Array{String,1},Dict{Any,Any},Set{Any}}, ::String) at ./file.jl:425
 [5] mktemp at ./file.jl:423 [inlined]
 [6] build! at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:635 [inlined] (repeats 2 times)
 [7] build(::Array{String,1}) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/entry.jl:647
 [8] #4 at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:36 [inlined]
 [9] cd(::getfield(Pkg.Dir, Symbol("##4#7")){Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},typeof(Pkg.Entry.build),Tuple{Array{String,1}}}, ::String) at ./file.jl:72
 [10] #cd#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:36
 [11] cd(::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/dir.jl:25
 [12] build(::String, ::Vararg{String,N} where N) at /home/paula/julia/usr/share/julia/stdlib/v0.7/Pkg/src/Pkg.jl:256
 [13] top-level scope

scope

e

**Comment by fredrikekre** [comment]
You can apply this patch
```diff
diff --git a/deps/kspec.jl b/deps/kspec.jl
index 685d250..21ab6be 100644
--- a/deps/kspec.jl
+++ b/deps/kspec.jl
@@ -41,7 +41,7 @@ function installkernel(name::AbstractString, julia_options::AbstractString...;
     juliakspec = joinpath(tempdir(), spec_name)
     try
         binary_name = Compat.Sys.iswindows() ? "julia.exe" : "julia"
-        kernelcmd_array = String[joinpath(JULIA_HOME,"$binary_name"), "-i",
+        kernelcmd_array = String[joinpath(Compat.Sys.BINDIR,"$binary_name"), "-i",
                                  "--startup-file=yes", "--color=yes"]
         append!(kernelcmd_array, julia_options)
         ijulia_dir = get(ENV, "IJULIA_DIR", dirname(@__DIR__)) # support non-Pkg IJulia installs
diff --git a/src/kernel.jl b/src/kernel.jl
index 7d09688..ea7a5f5 100644
--- a/src/kernel.jl
+++ b/src/kernel.jl
@@ -2,7 +2,7 @@ import IJulia
 
 # workaround #60:
 if IJulia.Compat.Sys.isapple()
-    ENV["PATH"] = JULIA_HOME*":"*ENV["PATH"]
+    ENV["PATH"] = Compat.Sys.BINDIR*":"*ENV["PATH"]
 end
 
 IJulia.init(ARGS)
```

**Comment by fredrikekre** [comment]
Will be fixed by #623 , closing as duplicate of #637
